artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/JsonUtil.java

```diff
@@ -295,6 +295,13 @@ else if (jsonValue == JsonValue.FALSE) {
             return jsonValue.toString();
          }
       }
+      else if (jsonValue instanceof Object[]) {
+         Object[] array = (Object[]) jsonValue;
+         for (int i = 0; i < array.length; i++) {
+            array[i] = convertJsonValue(array[i], desiredType);
```

Comment:
The use case for this that I've seen are management methods that return stuff like String[].  That said, I see your point.